test/Microsoft.IdentityModel.Tokens.Saml.Tests/SamlSerializerTests.cs

```diff
@@ -0,0 +1,308 @@
+                        SamlSerializer = new SamlSerializerPublic(),
+                        AudienceRestrictionConditionTestSet = ReferenceXml.SamlAudienceRestrictionConditionSingleAudience,
+                        TestId = nameof(ReferenceXml.SamlAudienceRestrictionConditionSingleAudience)
+                    },
+                    new SamlTheoryData
+                    {
+                        ExpectedException = ExpectedException.NoExceptionExpected,
+                        SamlSerializer = new SamlSerializerPublic(),
+                        AudienceRestrictionConditionTestSet = ReferenceXml.SamlAudienceRestrictionConditionMultiAudience,
+                        TestId = nameof(ReferenceXml.SamlAudienceRestrictionConditionMultiAudience)
+                    }
+                };
+            }
+        }
+        #endregion
+
+        #region SamlConditions
+#pragma warning disable CS3016 // Arrays as attribute arguments is not CLS-compliant
+        [Theory, MemberData("ReadConditionsTheoryData")]
+#pragma warning restore CS3016 // Arrays as attribute arguments is not CLS-compliant
+        public void ReadConditions(SamlTheoryData theoryData)
+        {
+            TestUtilities.WriteHeader($"{this}.ReadConditions", theoryData);
+            var context = new CompareContext($"{this}.ReadConditions, {theoryData.TestId}");
+            try
+            {
+                var sr = new StringReader(theoryData.ConditionsTestSet.Xml);
+                var reader = XmlDictionaryReader.CreateDictionaryReader(XmlReader.Create(sr));
+                var conditions = (theoryData.SamlSerializer as SamlSerializerPublic).ReadConditionsPublic(reader);
+                theoryData.ExpectedException.ProcessNoException();
+
+                IdentityComparer.AreEqual(conditions, theoryData.ConditionsTestSet.Conditions, context);
+            }
+            catch (Exception ex)
+            {
+                theoryData.ExpectedException.ProcessException(ex);
+            }
+
+            TestUtilities.AssertFailIfErrors(context);
+        }
+
+        public static TheoryData<SamlTheoryData> ReadConditionsTheoryData
+        {
+            get
+            {
+                return new TheoryData<SamlTheoryData>
+                {
+                    new SamlTheoryData
+                    {
+                        ExpectedException = ExpectedException.NoExceptionExpected,
+                        First = true,
+                        SamlSerializer = new SamlSerializerPublic(),
+                        ConditionsTestSet = ReferenceXml.SamlConditionsNoNbf,
+                        TestId = nameof(ReferenceXml.SamlConditionsNoNbf)
+                    },
+                    new SamlTheoryData
+                    {
+                        ExpectedException = ExpectedException.NoExceptionExpected,
+                        SamlSerializer = new SamlSerializerPublic(),
```

Comment:
Please keep properties in lexicographic order: ConditionsTestSet, SamlSerialzer, TestId.  Please adjust as needed.